lib/matplotlib/axes/_axes.py

```diff
@@ -6409,23 +6409,32 @@ def hist(self, x, bins=None, range=None, density=False, weights=None,
              orientation='vertical', rwidth=None, log=False,
              color=None, label=None, stacked=False, **kwargs):
         """
-        Plot a histogram.
+        Compute and plot a histogram.
 
-        Compute and draw the histogram of *x*.  The return value is a tuple
-        (*n*, *bins*, *patches*) or ([*n0*, *n1*, ...], *bins*, [*patches0*,
-        *patches1*, ...]) if the input contains multiple data.  See the
-        documentation of the *weights* parameter to draw a histogram of
-        already-binned data.
+        This method uses `numpy.histogram` to bin the data in *x* and count the
+        number of values in each bin, then draws the distribution either as a
+        `.BarContainer` or `.Polygon`. The *bins*, *range*, *density*, and
+        *weights* parameters are forwarded to `numpy.histogram`. If the data
+        has already been binned and counted, we recommend using `~.bar` or
+        `~.stairs` to plot the distribution::
 
-        Multiple data can be provided via *x* as a list of datasets
-        of potentially different length ([*x0*, *x1*, ...]), or as
-        a 2D ndarray in which each column is a dataset.  Note that
-        the ndarray form is transposed relative to the list form.
+            counts, bins = np.histogram(x)
+            plt.stairs(bins, counts)
 
-        Masked arrays are not supported.
+        To plot pre-computed bins and counts using ``hist()``, treat each bin
```

Comment:
This needs a transition with the previous paragraph (as it's another solution to the same problem), e.g. "Alternatively, one can also..." although I may not even bother showcasing that approach?